packages/channel/src/errors.ts

```diff
@@ -0,0 +1,29 @@
+import { CustomError } from "ts-custom-error";
+
+export class CannotWriteToFullBufferError extends CustomError {
```

Comment:
Not sure why we need two separate errors for “CannotWroteToFullBufferError” or “CannotReadFromEmptyBufferError”